### PR TITLE
Update to Crystal/0.25.0

### PR DESCRIPTION
**Pull request by robacarp**
This is an internal branch copy of the branch @bcardiff opened in #826

### CODE REVIEW

**Comment by robacarp** [comment]
This is _pretty close_. 

I've disabled automatic json parsing of json requests for now, I think that will need to be revisited and might even not be worth it for the framework to do.

A crystal spec produces this output, from build_spec:

```text

Failures:

  1) building a generated app generates a binary
     Failure/Error: File.exists?("bin/#{TEST_APP_NAME}").should be_true

       Expected: true
            got: false

     # /Users/robert/Documents/repositories/amberframework/amber/spec/build_spec.cr:45

  2) building a generated app crystal spec can be executed
     Failure/Error: spec_result.should contain "Finished in"

       Expected:   ""
       to include: "Finished in"

     # /Users/robert/Documents/repositories/amberframework/amber/spec/build_spec.cr:55

Finished in 19.56 seconds
106 examples, 2 failures, 0 errors, 0 pending

Failed examples:

crystal spec /Users/robert/Documents/repositories/amberframework/amber/spec/build_spec.cr:44 # building a generated app generates a binary
crystal spec /Users/robert/Documents/repositories/amberframework/amber/spec/build_spec.cr:54 # building a generated app crystal spec can be executed
```

It seems like there are a few places in generated apps which need to be updated.

**Comment by robacarp** [comment]
Status:

- [ ] Amber compiles
- [ ] Specs pass
- [ ] Generated app compiles
- [ ] Dependent shards are all up to date

Shard Hacks Needing Resolving:
- [ ] Citrine-i18n -> Techmagistar/i18n dependency chain needs fixed. [Pull request](https://github.com/TechMagister/i18n.cr/pull/6) Currently pointing to a fork kindly provided by @mamantoha
  - [ ] Citrine-i18n needs update PR, merge and release
- [ ] Garnet-Spec -> ysbaddaden/selenium-webdriver-crystal dependency chain needs fixed. [Pull request](https://github.com/ysbaddaden/selenium-webdriver-crystal/pull/24) Currently pointing to a fork kindly provided by @paulcsmith
  - [ ] Garnet-Spec needs update PR, merge and release
- [ ] Amber Router needs PR merge and release, [Pull request](https://github.com/amberframework/amber-router/pull/8)
- [ ] Granite needs PR merge and release, [Pull request](https://github.com/amberframework/granite/pull/231)

**Comment by robacarp** [comment]
Sorry, there was one glaring omission from my list above. JSON params parsing is completely disabled at the moment.

File: src/amber/router/parsers/json.cr [hunk 1]
```diff
@@ -2,24 +2,24 @@ module Amber::Router::Parsers
...
-                  json_params[key.as(String)] = value
-                else
-                  json_params[key.as(String)] = value.to_json
-                end
-              end
-            when Array
-              json_params["_json"] = json.to_json
-            end
+
+      return json_params unless request_body = request.body
```
Comment (eliasjpr) [inline src/amber/router/parsers/json.cr]: 👍

**Comment by ilovezfs** [comment]
Hurray! @faustinoaq can we get a new tag please? 🙏

**Comment by faustinoaq** [comment]
> can we get a new tag please

@ilovezfs Already working on that :+1:

We're trying to find critical bugs and fix them before a new release :wink:

**Pull request merged**